### PR TITLE
Add periphery meshing option in RGMB

### LINKED ISSUE #22846: Add periphery meshing option in RGMB
## Reason
The `Reactor` module contains periphery meshing capabilities provided by `PeripheralTriangleMeshGenerator` and `PeripheralRingMeshGenerator`. but are not integrated into the `CoreMeshGenerator` as part of the RGMB system.

## Design
This change adds the option to use either PTMG or PRMG as MeshSubgenerators to perform periphery meshing as part of the `CoreMeshGenerator` RGMB step. 

## Impact
This change adds core-periphery meshing functionality to the RGMB system.

### PR DESCRIPTION
**Pull request by aaronoaks**
closes #22846.

### CODE REVIEW

**Review by GiudGiud — CHANGES_REQUESTED** [review]
the files need to be moved to reactor/src/meshgenerator and same for include for the header

but I am seeing another version of this item already in MOOSE?
https://mooseframework.inl.gov/source/meshgenerators/PeripheralTriangleMeshGenerator.html

**Comment by aaronoaks** [comment]
@GiudGiud apologies, looks like some extra files were added to the commit, including the noted version of PTMG. These have been removed. This PR should only include changes for `CoreMeshGenerator`.

File: modules/reactor/doc/content/source/meshgenerators/CoreMeshGenerator.md
Comment (GiudGiud) [inline modules/reactor/doc/content/source/meshgenerators/CoreMeshGenerator.md:41]:
```suggestion
The `CoreMeshGenerator` includes support for meshing a circular reactor periphery surrounding the core. This integration supports using either [`PeripheralRingMeshGenerator`](PeripheralRingMeshGenerator.md) or [`PeripheralTriangleMeshGenerator`](PeripheralTriangleMeshGenerator.md), selected using the [!param](/Mesh/CoreMeshGenerator/periphery_generator) input option (by specifying either `PRMG` or `PTMG`, respectively). The input options for these mesh generators are provided below, but more details on their meaning and usage can be found in their respective documentation pages. The generated periphery region is given the block name [!param](/Mesh/CoreMeshGenerator/periphery_block_name) (default `outer_core`) and extra integer reporting ID `region_id` [!param](/Mesh/CoreMeshGenerator/periphery_region_id), along with outer boundary name "outside_periphery".
```

File: modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests
Comment (GiudGiud) [inline modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests:128]:
```suggestion
    requirement = 'The system shall generate a 2D hex core mesh with a reactor periphery meshed using a triangular mesh.'
```

File: modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests
Comment (GiudGiud) [inline modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests:135]:
```suggestion
    requirement = 'The system shall generate a 2D hex core mesh with a reactor periphery meshed using a quadrilateral element ring pattern.'
```

File: modules/reactor/test/tests/meshgenerators/core_mesh_generator/core_periphery_ptmg_vol.i
Comment (GiudGiud) [inline modules/reactor/test/tests/meshgenerators/core_mesh_generator/core_periphery_ptmg_vol.i:152]: this is a little too light for testing a mesh.

If you dont want an exodiff, this reporter will output more meta data
https://mooseframework.inl.gov/source/reporters/MeshInfo.html

File: modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests
Comment (GiudGiud) [inline modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests:138]: this file is on the heavier side because of the fields you are outputting.

if you add
cli_args = '--mesh-only'
you ll test the mesh, no extra field

File: modules/reactor/src/meshgenerators/CoreMeshGenerator.C
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/CoreMeshGenerator.C:65]:
```suggestion
      "Desired (local) triangle area as a function of x,y; omit to skip non-uniform refinement");
```

File: modules/reactor/include/meshgenerators/CoreMeshGenerator.h
Comment (GiudGiud) [inline modules/reactor/include/meshgenerators/CoreMeshGenerator.h:65]:
```suggestion
  /// Desired (local) triangle area as a function of (x,y)
```

File: modules/reactor/include/meshgenerators/CoreMeshGenerator.h
Comment (GiudGiud) [inline modules/reactor/include/meshgenerators/CoreMeshGenerator.h:52]:
```suggestion
  const subdomain_id_type _periphery_region_id;
```

File: modules/reactor/include/meshgenerators/CoreMeshGenerator.h
Comment (GiudGiud) [inline modules/reactor/include/meshgenerators/CoreMeshGenerator.h:53]:
```suggestion
  /// block name of periphery mesh elements
```

File: modules/reactor/include/meshgenerators/CoreMeshGenerator.h
Comment (GiudGiud) [inline modules/reactor/include/meshgenerators/CoreMeshGenerator.h:54]: actually this is not used?

File: modules/reactor/src/meshgenerators/CoreMeshGenerator.C
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/CoreMeshGenerator.C:38]: it d be a lot clearer to choose between 'triangle' and 'quad_ring' ?

File: modules/reactor/src/meshgenerators/CoreMeshGenerator.C
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/CoreMeshGenerator.C:71]: this will look on the final page. I would leave it inside the other group OR
change the name to Periphery Meshing: PTMG specific (Parameters is added auatomatically)

File: modules/reactor/src/meshgenerators/CoreMeshGenerator.C
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/CoreMeshGenerator.C:127]:
```suggestion
                 "Periphery region id must be specified when using periphery meshing.");
```

File: modules/reactor/src/meshgenerators/CoreMeshGenerator.C
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/CoreMeshGenerator.C:115]:
```suggestion
```

File: modules/reactor/src/meshgenerators/CoreMeshGenerator.C
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/CoreMeshGenerator.C:111]: some of these could be moved to defaults
then instead of isParamValid below, you would use isParamSetByUser 

the advantages are:
- a shorter constructor
- some of these could just actually have defaults
num_layers of 1 or block_name of outer_core are good defaults

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/CoreMeshGenerator.C:111]: https://mooseframework.inl.gov/source/meshgenerators/PeripheralRingMeshGenerator.html does have defaults

**Review by GiudGiud — CHANGES_REQUESTED** [review]
sorry for the delay, lots to review nearing the end of the year

File: modules/reactor/test/tests/meshgenerators/core_mesh_generator/core_periphery_ptmg_vol.i
Comment (aaronoaks) [reply modules/reactor/test/tests/meshgenerators/core_mesh_generator/core_periphery_ptmg_vol.i:152]: This test was based on issues we had with testing refinement with PTMG itself, where the refinement produced slightly different meshes in different build envs (consistently Intel Mac didn't match the gold exodus files in the CIVET tests):

https://github.com/idaholab/moose/pull/22436#issuecomment-1285874705

I'm assuming using the mesh_info reporter will have a similar outcome, because they won't be exactly the same mesh.

Comment (GiudGiud) [reply modules/reactor/test/tests/meshgenerators/core_mesh_generator/core_periphery_ptmg_vol.i:152]: ok sounds good. Doubt we can do better

**Comment by GiudGiud** [comment]
Let s squash all the non important formatting commits together please

I raised a question about line endings issues on @miaoyinb 's PR I d like to know what s going on before we merge more code

**Comment by aaronoaks** [comment]
All of the commits have been squashed to a single commit.

**Review by GiudGiud — APPROVED** [review]
Looks great!

Merges are frozen right now due to some deployment issues, we ll merge this in once this is unlocked

**Pull request merged**
